modules/iam-assumable-role-with-oidc/main.tf

```diff
@@ -32,7 +33,7 @@ data "aws_iam_policy_document" "assume_role_with_oidc" {
       condition {
         test     = "ArnLike"
         variable = "aws:PrincipalArn"
-        values   = ["arn:${local.partition}:iam::${local.account_id}:role${var.role_path}${local.role_name_condition}"]
+        values   = ["arn:${local.partition}:iam::${data.aws_caller_identity.current.account_id}:role${var.role_path}${local.role_name_condition}"]
```

Comment:
Revert this change.

Reply:
Could you please elaborate on why it's better to use data instead of local?

Reply:
Usually, do not touch it if it works is a very important rule. Such microchanges pollute the PR and make it hard to review. This PR is doing more than fixing an original issue.

Introducing `locals` makes perfect sense when there is some logic in it (e.g. `local.aws_account_id`).

Reply:
Got it, thank you for the explanation.

Reply:
@antonbabenko  This one can't be reverted because the module will stop working due to missing `account_id` local. If you check locals above you can see that `aws_account_id` is declared, however, `account_id` is not. When I looked at [this](https://github.com/terraform-aws-modules/terraform-aws-iam/pull/283/files#diff-ab579ba376417161fb0950cae991b5ffb34ee5c20c780f0cc154ad5fbf263880L28) change which introduced the problem it seemed obvious to just revert it to state it was before.